mangaki/mangaki/utils/lasso.py

```diff
@@ -0,0 +1,63 @@
+from mangaki.utils.common import RecommendationAlgorithm
+from mangaki.settings import DATA_DIR
+from scipy.sparse import coo_matrix, load_npz
+from sklearn.linear_model import Lasso
+from sklearn.preprocessing import scale
+from collections import Counter, defaultdict
+from mangaki.utils.stats import avgstd
+import os.path
+
+
+def relu(x):
+    return max(-2, min(2, x))
+
+
+class MangakiLASSO(RecommendationAlgorithm):
+    def __init__(self, with_bias=True, alpha=0.01):
+        super().__init__()
+        self.alpha = alpha
+        self.with_bias = with_bias
+
+    def load_tags(self, T=None):
+        if T is None:
+            T = load_npz(os.path.join(DATA_DIR, 'balse/tag-matrix.npz')).tocsc()
+            T_scaled = scale(T, with_mean=False).toarray()
+        _, self.nb_tags = T.shape
+        self.T = T_scaled
+
+    def fit(self, X, y):
+        self.load_tags()
+        row = X[:, 0]
+        col = X[:, 1]
+        self.nb_rated = Counter(col)
+        data = y
+        M = coo_matrix((data, (row, col)), shape=(self.nb_users, self.nb_works)).tocsr()
+        self.reg = defaultdict(lambda: Lasso(alpha=self.alpha, fit_intercept=self.with_bias))
```

Comment:
Optimally, the `self.reg` should be frozen into a `dict` afterwards so that no one else than `fit` create `Lasso` instances without knowing it. (easy mistake, just try to access the `coef_` of any non-existent user ID, it'll be confusing.)

Reply:
Done.